src/plugins/index_pattern_field_editor/public/components/preview/field_preview_context.tsx

```diff
@@ -92,6 +98,9 @@ const defaultParams: Params = {
   format: null,
 };
 
+export const defaultValueFormatter = (value: unknown) =>
```

Comment:
You should copy how index patterns does this - `src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts` - `getFormatterForField`

Reply:
It is a slightly different scenario. `getFormatterForField`  works with field "types"

```js
return this.fieldFormats.getDefaultInstance(
  field.type as KBN_FIELD_TYPES,
  field.esTypes as ES_FIELD_TYPES[]
);
```

Here I only want to warp the value with a `<span>` if the formatter is not found (and don't want to over engineer it either 😊 )

Reply:
I'll leave it up to you to judge the tradeoffs but doing things the same way as index patterns will increase the likelihood that everything will stay in sync a bit better. You have the elasticsearch field type and can use `castEsToKbnFieldTypeName` to get the kibana field type. From  `src/plugins/data/common/kbn_field_types/kbn_field_types.ts`

Reply:
Yes, I am probably missing some bit of information here and maybe better to discuss over zoom. Here the only thing I try to do is: given a formatter (with its `id`) and a `value`, format the value. If there are no formatter with the provided `id` just warp the value with a `<span>`. I am not sure when field types enter into play tbh.